extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -189,168 +227,54 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	getChildren(element?: BookTreeItem): Thenable<BookTreeItem[]> {
 		if (element) {
 			if (element.sections) {
-				return Promise.resolve(this.getSections(element.tableOfContents, element.sections, element.root));
+				return Promise.resolve(this.currentBook.getSections(element.tableOfContents, element.sections, element.root));
 			} else {
 				return Promise.resolve([]);
 			}
 		} else {
-			return Promise.resolve(this.getBooks());
+			let booksitems: BookTreeItem[] = [];
+			this.books.map(book => {
+				booksitems = booksitems.concat(book.bookItems);
+			});
+			return Promise.resolve(booksitems);
 		}
 	}
 
-	/**
-	 * Recursively parses out a section of a Jupyter Book.
-	 * @param array The input data to parse
-	 */
-	private parseJupyterSection(array: any[]): IJupyterBookSection[] {
-		return array.reduce((acc, val) => Array.isArray(val.sections) ? acc.concat(val).concat(this.parseJupyterSection(val.sections)) : acc.concat(val), []);
-	}
 
-	public async getBooks(): Promise<BookTreeItem[]> {
-		let books: BookTreeItem[] = [];
-		for (const contentPath of this._tableOfContentPaths) {
-			let root = path.dirname(path.dirname(contentPath));
-			try {
-				const config = yaml.safeLoad((await fs.readFile(path.join(root, '_config.yml'), 'utf-8')).toString());
-				const tableOfContents = yaml.safeLoad((await fs.readFile(contentPath, 'utf-8')).toString());
-				try {
-					let book = new BookTreeItem({
-						title: config.title,
-						root: root,
-						tableOfContents: { sections: this.parseJupyterSection(tableOfContents) },
-						page: tableOfContents,
-						type: BookTreeItemType.Book,
-						treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Expanded,
-					},
-						{
-							light: this._extensionContext.asAbsolutePath('resources/light/book.svg'),
-							dark: this._extensionContext.asAbsolutePath('resources/dark/book_inverse.svg')
-						}
-					);
-					books.push(book);
-				} catch (e) {
-					throw Error(localize('invalidTocError', "Error: {0} has an incorrect toc.yml file. {1}", config.title, e instanceof Error ? e.message : e));
-				}
-			} catch (e) {
-				let error = e instanceof Error ? e.message : e;
-				this._errorMessage = error;
-				vscode.window.showErrorMessage(error);
+	getParent(element?: BookTreeItem): vscode.ProviderResult<BookTreeItem> {
+		let parentPath;
+		if (element.root.endsWith('.md')) {
+			parentPath = path.join(this.currentBook.bookPath, 'content', 'readme.md');
```

Comment:
The param is defined as optional so you should be checking that element isn't undefined before accessing its properties (same for element.root if that's also possibly undefined)